src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java

```diff
@@ -334,6 +334,21 @@ public void onDeleted(Run run) {
                 return;
             }
 
+            // If the build already complete, this could be a Jenkins cleanup operation
+            if (buildData.isCompleted()) {
```

Comment:
Note: updated this clause to check for the underlying `completeBuild` flag instead of `null`.  See the breakdown of what's considered complete in the [Result class](https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/model/Result.java).